please/please.py

```diff
@@ -11,172 +11,192 @@
+@app.command(short_help="Mark a task as done")
+def done(index: int) -> None:
     index = index - 1
-    if index >= len(config["tasks"]):
-        center_print_wrap(
-            "Are you sure you gave me the correct number to mark as done?", "bright_red on bright_white")
-        return
 
-    if len(config["tasks"]) > 0 and not all_tasks_done():
+    if len(config["tasks"]) == 0:
+        center_print_wrap(
+            "Task list is empty",
+            style="bright_red on bright_white",
+        )
+    elif all_tasks_done():
+        center_print("All tasks are already completed!", "black on green")
+    elif not 0 <= index < len(config["tasks"]):
+        center_print_wrap(
+            "Are you sure you gave me the correct number to mark as done?",
+            "bright_red on bright_white",
+        )
+    else:
         config["tasks"][index]["done"] = True
         write_config(config)
         center_print("Updated Task List", "black on green")
         print_tasks(config["tasks"])
-    else:
-        center_print_wrap(
-            "Sorry, I've got no tasks to mark as done", style="bright_red on bright_white")
 
 
-@app.command(short_help='Mark a task as undone')
-def undone(index: int):
+@app.command(short_help="Mark a task as undone")
+def undone(index: int) -> None:
     index = index - 1
-    if index >= len(config["tasks"]):
+    if len(config["tasks"]) == 0:
         center_print_wrap(
-            "Are you sure you gave me the correct number to mark as undone?", "bright_red on bright_white")
-        return
+            "Task list is empty",
+            style="bright_red on bright_white",
+        )
 
-    if len(config["tasks"]) > 0:
+    elif not 0 <= index < len(config["tasks"]):
+        center_print_wrap(
+            "Are you sure you gave me the correct number to mark as undone?",
+            "bright_red on bright_white",
+        )
+    else:
         config["tasks"][index]["done"] = False
         write_config(config)
         center_print("Updated Task List", "black on green")
         print_tasks(config["tasks"])
-    else:
-        center_print_wrap(
-            "Sorry, I've got no tasks to mark as undone", style="bright_red on bright_white")
 
 
 @app.command(short_help="Show all Tasks")
-def showtasks():
-    tasks_list = config["tasks"]
-    if len(tasks_list) > 0:
-        table1 = Table(title="Tasks", title_style="grey39",
-                       header_style="#e85d04",  style="#e85d04 bold")
-        table1.add_column('Number', style="#e85d04")
-        table1.add_column('Task')
-        table1.add_column('Status')
-
-        for index, task in enumerate(tasks_list):
-            task_name = f"""[#A0FF55]{task["name"]}[/]""" if task["done"] else f"""[#FF5555]{task["name"]}[/]"""
-            task_status = "✅" if task["done"] else "❌"
-            task_index = "[#A0FF55]" + str(
-                index + 1) + "[/]" if task["done"] else "[#FF5555]" + str(index + 1) + "[/]"
+def showtasks() -> None:
```

Comment:
We do not want to show anything at all when tasks table is empty. This has to be reverted, sorry.

Reply:
the program exits without any cue if we call `showtasks()` when the task table is empty. This might confuse users (it confused me when it happened for the first time)
I think informing that table is empty is better than simply exiting.

Reply:
> the program exits without any cue if we call `showtasks()` when the task table is empty. This might confuse users (it confused me when it happened for the first time) I think informing that table is empty is better than simply exiting.

Oh, You're right. I think the way we'd like to do it, is to show "Looking good, no pending tasks" when showing the default screen and when the table is empty and someone still calls showtasks, the user should get "Sorry, I've got no tasks to show". Thanks!